README.md

```diff
@@ -1,29 +1,369 @@
-# TagoIO Toolkit to build your own widgets
-Create your amazing widgets and run it inside TagoIO admin.
+# TagoIO Custom Widget SDK
 
-### How to install?
+Create amazing custom widgets and run them inside TagoIO Admin with this comprehensive toolkit. The Custom Widget SDK provides a powerful JavaScript interface for building interactive IoT dashboards and data visualization components.
+
+## 🚀 Quick Start
+
+### CDN Installation (Recommended for beginners)
+
+The simplest way to get started is to include the SDK directly in your HTML:
 
-The simplest way to use is to add the <script> direct on your project.
 ```html
-<script src="https://admin.tago.io/dist/custom-widget.min.js"></script>
-<link rel="stylesheet" href="https://admin.tago.io/dist/custom-widget.min.css"> <!-- OPTIONAL -->
+<!DOCTYPE html>
+<html>
+<head>
+    <title>My Custom Widget</title>
+    <script src="https://admin.tago.io/dist/custom-widget.min.js"></script>
+    <link rel="stylesheet" href="https://admin.tago.io/dist/custom-widget.min.css"> <!-- OPTIONAL -->
+</head>
+<body>
+    <script>
+        // Your widget code here
+        window.TagoIO.onStart(function(widget) {
+            console.log('Widget started!', widget);
+        });
+        
+        window.TagoIO.ready();
+    </script>
+</body>
+</html>
 ```
 
-Also, you can add it on your project to use with webpack
+### NPM Installation (For development environments)
+
+For projects using webpack or other build tools:
+
 ```bash
-$ npm install @tago-io/custom-widget --save
+npm install @tago-io/custom-widget --save
 ```
 
-On top of your entry component
+Then import in your entry component:
+
 ```javascript
 import "@tago-io/custom-widget";
 import "@tago-io/custom-widget/dist/custom-widget.css"; // OPTIONAL
 ```
 
-### Projects Examples
+## 📚 SDK Documentation
+
+### Core API Reference
+
+The TagoIO Custom Widget SDK provides a comprehensive API through the global `window.TagoIO` object:
+
+#### Initialization Functions
+- **`TagoIO.ready(options)`** - Signal that your widget is ready to start
+- **`TagoIO.onStart(callback)`** - Handle widget startup and receive configuration
+- **`TagoIO.onError(callback)`** - Handle API errors and widget issues
+
+#### Data Operations
+- **`TagoIO.sendData(data, callback)`** - Send data to TagoIO devices
+- **`TagoIO.editData(data, callback)`** - Edit existing device data
+- **`TagoIO.deleteData(data, callback)`** - Delete device data
+- **`TagoIO.editResourceData(data, callback)`** - Edit platform resources
+
+#### Real-time Data
+- **`TagoIO.onRealtime(callback)`** - Receive real-time data updates
+- **`TagoIO.onSyncUserInformation(callback)`** - Access user context and authentication
+- **`TagoIO.onSyncBlueprintDevices(callback)`** - Access blueprint device configurations
```

Comment:
The explanation is too short for the impact of the Realtime events. You must state those are async events that triggers based on Dashboard realtime events. For instance, `onRealtime` its called when new data is received through realtime events.